### ISSUE TITLE
rds_instance reports changed despite no changes to the database

### ISSUE THREAD

**Issue by skoranda** [ansible-collections/community.aws#240]
##### SUMMARY

community.aws.rds_instance reports changed despite no changes to the database

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

community.aws.rds_instance

##### ANSIBLE VERSION

```paste below
ansible 2.9.6
```

##### CONFIGURATION

```paste below
ANSIBLE_SSH_ARGS(env: ANSIBLE_SSH_ARGS) = -F /home/skoranda/CILogon2/comanage-registry-ansible/ssh_config -C -o ControlMaster=auto -o ControlPersist=3600s
ANSIBLE_SSH_CONTROL_PATH(/home/skoranda/CILogon2/comanage-registry-ansible/ansible.cfg) = ssh_mux_%%h_%%p
DEFAULT_HOST_LIST(env: ANSIBLE_INVENTORY) = ['/home/skoranda/CILogon2/comanage-registry-ansible/aws_ec2.yml']
DEFAULT_TIMEOUT(/home/skoranda/CILogon2/comanage-registry-ansible/ansible.cfg) = 60
DEFAULT_VAULT_PASSWORD_FILE(env: ANSIBLE_VAULT_PASSWORD_FILE) = /home/skoranda/.vault_pass.txt
INTERPRETER_PYTHON(/home/skoranda/CILogon2/comanage-registry-ansible/ansible.cfg) = auto_silent
```

##### OS / ENVIRONMENT

Debian 10.5

##### STEPS TO REPRODUCE

Run a playbook with the following task:

```yaml
    - name: Provision DB 
      community.aws.rds_instance:
        availability_zone: us-east-2a
        engine: aurora
        engine_version: 5.6.10a
        allow_major_version_upgrade: no
        auto_minor_version_upgrade: yes
        db_cluster_identifier: cilogon-registry-cluster-01-dev
        db_instance_class: db.t3.small
        db_instance_identifier: cilogon-registry-dev
        db_parameter_group_name: cilogon-aurora-mysql-5-6-parameter-group-dev
        state: present
        region: us-east-2
```
Wait for the database to be created.

Run the same playbook with the same task. The task shows

```paste below
TASK [Provision DB] *********************************************************************
[WARNING]: Module did not set no_log for force_update_password
changed: [localhost]
```
even though nothing has changed.

##### EXPECTED RESULTS

After the database is created and no changes are made to the task I expect running the task to show 

```
ok: [localhost]
```

##### ACTUAL RESULTS

```paste below
TASK [Provision DB] *********************************************************************
[WARNING]: Module did not set no_log for force_update_password
changed: [localhost]
```

**Comment by skoranda** [comment]
I also verified that the issue happens with 2.10.0 and 2.11.0.dev0

**Comment by tremble** [comment]
This issue should have been fixed by #1002 which is available in release 3.3.0
